tokio-trace/tokio-trace-core/src/callsite.rs

```diff
@@ -67,25 +94,24 @@ pub struct Identifier(
     pub &'static Callsite,
 );
 
+/// Clear and reregister interest on all callsites
+pub fn recache_interest() {
+    let mut registry = REGISTRY.lock().unwrap();
+    registry.reestablish();
+}
+
 /// Register a new `Callsite` with the global registry.
 ///
 /// This should be called once per callsite after the callsite has been
 /// constructed.
 pub fn register(callsite: &'static Callsite) {
     let mut registry = REGISTRY.lock().unwrap();
-    let meta = callsite.metadata();
-    registry.dispatchers.retain(|registrar| {
-        match registrar.try_register(meta) {
-            Some(interest) => {
-                callsite.add_interest(interest);
-                true
-            }
-            // TODO: if the dispatcher has been dropped, should we invalidate
-            // any callsites that it previously enabled?
-            None => false,
-        }
-    });
+    let dispatchers_changed = registry.establish_interest(callsite);
     registry.callsites.push(callsite);
+
+    if dispatchers_changed {
```

Comment:
yea if a new callsite registers after a dispatcher has been dropped but before anyone else has checked for the dropped dispatcher it would set the callsite, then drop the dispatcher and recache the callsite along with all others.

Reply:
I pushed the change to move the push after the reestablish.

Reply:
Okay, great!